pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +423,51 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+// AttachAndMount creates and attaches a vmdk as a non-persistent disk, mounts it, and returns the mount path.
+func (m *Manager) AttachAndMount(op trace.Operation, datastoreURI *object.DatastorePath) (string, error) {
+	config := NewNonPersistentDisk(datastoreURI)
+	d, err := m.CreateAndAttach(op, config)
```

Comment:
@caglar10ur was this the call pattern you had in mind for non-persistent attach of existing disk? While it does technically create a new delta disk from  our perspective it's not a new disk.